pandas/io/excel/_openpyxl.py

```diff
@@ -437,10 +437,12 @@ def write_cells(
                         f"Sheet '{sheet_name}' already exists and "
                         f"if_sheet_exists is set to 'error'."
                     )
+                elif self.if_sheet_exists == "overlay":
+                    wks = self.sheets[sheet_name]
```

Comment:
Is the error message by openpyxl descriptive enough if the sheet doesn't exist or would it be better to raise a more descriptive error?

Reply:
There cannot be an error because of line 427:
```
if sheet_name in self.sheets and self.if_sheet_exists != "new":
```
unless someone has removed the sheet themselves  by editing `writer.book` with openpyxl functions. This is currently not allowed by he documentation. #43088 is about this